### DISCUSSION TITLE
Will 14.3.0 be ever released?

### DISCUSSION THREAD

**Discussion by max-mayorov** [vercel/next.js#66369] [Help]
### Summary

Hi,

We are waiting for a while for 14.3.0 release because it contains bug fixes that are important for us.
However, in the releases I see that after 14.3.0-canary it has immediately jumped onto 15.0.0-canary.
![image](https://github.com/vercel/next.js/assets/4355960/77054999-2133-4d4a-b590-e708dff3947c)

What does that mean? Will 14.3.0 ever be released? If yes, is there a date planned already?

Thanks!
Max

**Comment by vlucas** [comment]
I am also looking for a 14.3 release since it fixes a memory leak in `fetch` that we have been dealing with in production for about 6 months now. It would be nice to have an official release that doesn't also do a major React version bump that may require more changes in the codebase.

The issue was catalogued here: https://github.com/vercel/next.js/issues/64212

And then fixed in this PR: https://github.com/vercel/next.js/pull/64746

**Reply by max-mayorov** [reply]
Yes, nextjs releases are a mess :( Actually, we are on 14.1.2 and we can't upgrade at all because 14.2.3 breaks the behavior of `[...catchAll]` in parallel routes 👎... Frustrated...

**Comment by yarick2000** [comment]
Same for us, waiting 14.3 for the same reason...

**Comment by m-sanders** [comment]
It would be good to get the fixes too. I’ve always found that it takes a while for all the deps to upgrade and catch up when a major release comes out. I am really hoping that nextjs at least has some overlap and back ports fixes…

**Comment by kamilgrzegorczyk** [comment]
I would like to learn that answer too as it seems all the work is now getting pushed into the 15.x branch

**Reply by yarick2000** [reply]
Btw, I see that they've switched to React 19RC in 14.3 canaries... I'm wondering why. I was hoping they will stick to React 18 in this version...

**Reply by kamilgrzegorczyk** [reply]
I would think the same - that such breaking change as bumping up to 19.x should happen in the major version of next, not between minors.

**Reply by furai** [reply]
Still no news from Vercel?

**Comment by erwanriou** [comment]
For me it sound like the issue scope was too much and they lately decided to do a major release instead of a minor one due to too much breaking changes?

Am i correct in my theory?

**Reply by jthrilly** [reply]
I think you are, but that is frankly ridiculous given the sheer volume of small fixes, and the fact that there hasn't been a point release since April.

Proper development practice would be to keep the breaking changes and the patches separate, so that we can have a patch release without having a major version bump of React.

Yet another illustration (as if we needed one) of how you cannot depend on NextJS at the moment.

**Comment by jthrilly** [comment]
I posted above, but I think the total number of comments is probably one of the metrics the nextJS team monitors when deciding to reply. Apologies if this is considered spam, but I want to add my voice.

It is **absolutely essential** for the future of this project that Nextjs, and React together adopt a reliable, consistent, and sane release schedule, and associated set of release practices. 

Developers make technology choices based on more than just features - we need to know that our framework will receive fixes and support. A new major version every year is great! We enjoy getting a taste of the future with canary builds, even if I do personally feel like you've gone way too far with pushing unstable and non-final APIs. But not everyone is building short term projects and YouTube todo apps that don't need to be maintained. We need bug fixes and non-breaking changes to be back ported to the most recent stable major release. This is basic stuff, and the lack of this is seriously eroding developer confidence. 

14.3.0 contained literally hundreds of small fixes and enhancements that were non breaking. These fixes were committed weeks and sometimes months ago. Someone somewhere made the absurd decision to bundle these changes with a set of hugely disruptive breaking changes including a major version bump of the core React dependency. Surely, _surely_, someone in the core team can see the issue with this?

Please, with absolute kindness, reconsider this decision. Release 14.3.0, and commit to properly maintaining the current stable version of the framework in parallel with whatever you are doing in canary 🙏

**Reply by max-mayorov** [reply]
I totally double that! We had a lots of pain using Nextjs 14 because of release hell. It even led us to a point then we had to deliver product on canary release for UAT because we can't wait for the point release anymore...

**Reply by develos-dr** [reply]
Im wondering now did my decision to use NEXT instead of NUXT was good -.-

**Reply by max-mayorov** [reply]
> Im wondering now did my decision to use NEXT instead of NUXT was good -.-

Well Next vs Nuxt is more React vs Vue question. I find React's functional approach more flexible and easier to understand than Vue's templating. But that's a really long discussion and an offtopic, anyways :D

I think it should be more Next vs Remix (?) question...

**Comment by develos-dr** [comment]
Agreed! Vue templating syntax is bad BUT when you make all in typescript with jsx syntax then it looks pretty like react 😃


Best Regards,



[Image]

Dawid Ratajczak

CEO

Mobile: +48 502-109-665
NIP: 604-022-87-29
REGON: 52121310300000
http://www.develos.pl/



Develos Ratajczak Gajos Spółka Komandytowo-Akcyjna, ul. Biała 4, 80-180 Jankowo Gdańskie
zarejestrowana w Sądzie Rejonowym Gdańsk-Północ w Gdańsku, VII Wydział Gospodarczy nr KRS: 0000952438.
Regon: 52121310300000 | Kapitał zakładowy: 50.000,00 zł | NIP: 604-022-87-29 | Nr konta bankowego: PLN 73 1140 1065 0000 5708 3500 1001 | USD: PL19 1140 1065 0000 5708 3500 1003 | EUR: PL46 1140 1065 0000 5708 3500 1002

Uwaga: Ten e-mail (wraz z załącznikami) jest poufny i przeznaczony tylko dla adresata(ów) tej wiadomości. Jeżeli nie jesteś adresatem niniejszej wiadomości, jej rozpowszechnianie, kopiowanie oraz rozprowadzanie(dystrybucja) jest zabronione. Jeśli wiadomość ta znalazła się w Twojej skrzynce w wyniku błędu, proszę poinformuj mnie o tym niezwłocznie i usuń oryginał wiadomości wraz z wszelkimi wydrukami(kopiami).

Important Note: This e-mail (including any attachments) is confidential and intended only for the use of the addressee (s) named herein. If you have received this e-mail in error, you are hereby notified that any review, copying or distribution of it is strictly prohibited. Please inform me immediately and destroy the original e-mail and any printouts.

________________________________
Od: Max Mayorov ***@***.***>
Wysłane: Tuesday, June 11, 2024 12:12:31 PM
Do: vercel/next.js ***@***.***>
DW: Dawid Ratajczak ***@***.***>; Comment ***@***.***>
Temat: Re: [vercel/next.js] Will 14.3.0 be ever released? (Discussion #66369)


Im wondering now did my decision to use NEXT instead of NUXT was good -.-

Well Next vs Nuxt is more React vs Vue question. I find React's functional approach more flexible and easier to understand than Vue's templating. But that's a really long discussion and an offtopic, anyways :D

**Comment by kamilgrzegorczyk** [comment]
It seems that 14.2.4 got released with some patches being backported. From our perspective though that is not affecting us too much (though I understand these patches may help a lot of ppl)

**Reply by AlexanderYW** [reply]
Thanks for sharing!

I was hoping it contained #53608, but it doesn't since that is categorized as a feature/change and not a bug fix. Which now pushes me to move away from using 'next/font'

**Reply by sydtv** [reply]
I really hope that they backport https://github.com/vercel/next.js/pull/65006 , we are not able to debug our app in the current state

**Reply by furai** [reply]
We need more things backported.

**Reply by mengxi-ream** [reply]
> Thanks for sharing!
> 
> I was hoping it contained #53608, but it doesn't since that is categorized as a feature/change and not a bug fix. Which now pushes me to move away from using 'next/font'

exactly same

**Reply by noahhai** [reply]
Same - I tried upgrading to 15 canary releases (need this [https://github.com/vercel/next.js/pull/68835](#68835) but ran into way too many compatibility issues with various libraries that don't support react 19 RC yet.

**Comment by webYouth** [comment]
14.3.0 hasn't been released yet
